pkg/model/stack.go

```diff
@@ -185,6 +193,14 @@ func (s *Stack) validate() error {
 		return fmt.Errorf("Invalid stack: 'services' cannot be empty")
 	}
 
+	for _, endpoints := range s.Endpoints {
+		for _, endpoint := range endpoints {
+			if !s.hasService(endpoint.Service) {
```

Comment:
You can do `if _, ok := s.Services[endpoint.Service]; !ok`

Reply:
Could you add tests for this logic?
also, check that the target port of the endpoint is defined in the service.